library/item/item_group/src/main/java/org/quiltmc/qsl/item/group/api/QuiltItemGroup.java

```diff
@@ -217,6 +236,18 @@ public Builder icon(@NotNull Supplier<ItemStack> iconSupplier) {
 			return this;
 		}
 
+		/**
+		 * This is used to add an icon to the {@link ItemGroup}.
+		 *
+		 * @param icon
+		 * @return {@code this}
+		 */
+		public Builder icon(@NotNull ItemGroupIcon icon) {
```

Comment:
Does this work on a dedicated server? I'm a bit wary of it causing loading of MatrixStack (whether by nature or when someone tries to use it), which as far as I'm aware doesn't exist on the server.